### ISSUE TITLE
Unexpected Replication Guide behavior w/ chained operators

### ISSUE THREAD

**Issue by AndrewHeumann** [DynamoDS/Dynamo#7934]
When chaining operators in a single expression in DesignScript, replication guide index value is ignored in the structuring of the output list. I suspect this is because Replication Guides are evaluated on a per-function basis - but seems like a missed opportunity to have these expressions be evaluated in an order that respects RG index. I would expect the above example to behave just like the one below, where reversing the RG values should flip the organization of the result:
![img](http://i.imgur.com/9o1DNJU.png)

**Comment by jnealb** [comment]
@ke-yu @Racel Ping.

**Comment by andydandy74** [comment]
@AndrewHeumann - I've come across this as well.
It'll be fine as long as you put all vars that have replication guides into parentheses:
![textreplication](https://user-images.githubusercontent.com/3014437/26843571-01b100f0-4af1-11e7-984f-5885601d8804.PNG)

**Comment by AndrewHeumann** [comment]
@andydandy74 yes I came upon that solution myself, which is fine - you're using parens to force the order of operations. My point is that it would be nice if the OOO respected replication guides in the first place.

**Comment by andydandy74** [comment]
No argument from me 😉
